components/Util/NumericInput.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState, Dispatch, SetStateAction } from 'react'
+
+export default function NumericInput({
+  label,
+  name,
+  max,
+  min,
+  step,
+  defaultValue,
+  value,
+  onChange,
+  onKeyDown,
+}: {
+  label: string
+  name?: string
+  max?: number
+  min?: number
+  step?: number
+  defaultValue?: number
+  value?: number
+  onChange?: Dispatch<SetStateAction<number>>
+  onKeyDown?: (value: number | undefined) => void
+}) {
+  // The logical operation here simulates XOR operation
+  // Only either of them should be TRUE at one time.
+  // This operation ensures that
+  if ((value && onChange) === defaultValue) {
+    throw new Error('Provide only either of value or defaultValue.')
+  }
+
+  const isControlled = value !== undefined && onChange !== undefined
+
+  const [internalValue, setInternelValue] = useState(
```

Comment:
internel -> internal

Reply:
Addressed in cc34c09e6c252ca8d298e44167934c24bd670b70
Thanks!